### PR TITLE
make linkcheck build with no errors

### PR DESCRIPTION
**Pull request by willingc**
- update dead links
- convert links to .md to .html
- remove anchor that no longer appears to exist in document registry
- correct commenting style in one javascript code block

cc / @jasongrout @jzf2101

### CODE REVIEW

**Comment by jasongrout** [comment]
Thanks!

**Comment by jasongrout** [comment]
Looks like appveyor is failing because now we finally have terminals available in windows, with the notebook 5.3 release today.

**Comment by jasongrout** [comment]
Merging so we can keep working on the docs.

File: docs/source/developer/virtualdom.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ In situations where these feature are needed, we recommend using Phosphor's
 `Widget` class directly.
 
 The idea of virtual DOM rendering, which became popular in the
-[React](https://facebook.github.io/react/) community, is a very elegant and
+[React](https://nodejs.org/en/) community, is a very elegant and
```
Comment (vidartf) [inline docs/source/developer/virtualdom.md]: This link change seems strange.

Comment (willingc) [reply docs/source/developer/virtualdom.md]: @vidartf Yeah, I just grabbed the redirect that facebook has currently from the old link as it displayed in `make linkcheck`. https://nodejs.org seems to also redirect to https://nodejs.org/en. Either is fine. Probably best to go with https://nodejs.org unless it errors in `make linkcheck`.

Comment (vidartf) [reply docs/source/developer/virtualdom.md]: The redirect in my browser goes to `https://reactjs.org/` not `https://nodejs.org/` ;)

Comment (vidartf) [reply docs/source/developer/virtualdom.md]: (at least currently, I'm not saying fb cannot have had an errant redirect 😅)

Comment (willingc) [reply docs/source/developer/virtualdom.md]: You are completely right @vidartf. 👍 I was working too fast. This should absolutely be react 🤦‍♀️. 

@jasongrout @jzf2101 Can you fix this the next time you touch the docs?

Comment (jasongrout) [reply docs/source/developer/virtualdom.md]: @vidartf already fixed it in https://github.com/jupyterlab/jupyterlab/pull/3667 :).

Comment (willingc) [reply docs/source/developer/virtualdom.md]: @vidartf Thank you :sunny:

**Pull request merged**